src/metabase/driver/teradata.clj

```diff
@@ -0,0 +1,306 @@
+  (cond
+    (true? value)  1
+    (false? value) 0
+    :else          value))
+
+(defn- get-tables
+  "Fetch a JDBC Metadata ResultSet of tables in the DB, optionally limited to ones belonging to a given schema."
+  ^ResultSet [^DatabaseMetaData metadata, ^String schema-or-nil]
+  (jdbc/result-set-seq (.getTables metadata nil schema-or-nil "%" ; tablePattern "%" = match all tables
+                         (into-array String ["TABLE", "VIEW", "FOREIGN TABLE"]))))
+
+(defn- fast-active-tables
+  "Teradata, fast implementation of `fast-active-tables` to support inclusion list."
+  [driver, ^DatabaseMetaData metadata, {{:keys [dbnames]} :details, :as database}]
+  (let [all-schemas (set (map :table_schem (jdbc/result-set-seq (.getSchemas metadata))))
+        dbs (dbnames-set dbnames)
+        schemas     (if (empty? dbs)
+                      (set/difference all-schemas excluded-schemas) ; use default exclusion list
+                      (set/intersection all-schemas dbs))] ; use defined inclusion list
+    (set (for [schema schemas
+               table-name (mapv :table_name (get-tables metadata schema))]
+           {:name   table-name
+            :schema schema}))))
+
+(defn- describe-database
+  "Overridden to have access to the database with the configured property dbnames (inclusion list)
+   which will be used to filter the schemas."
+  [driver database]
+  (sql/with-metadata [metadata driver database]
+    {:tables (fast-active-tables, driver, ^DatabaseMetaData metadata, database)}))
+
+(defn- run-query
+  "Run the query itself without setting the timezone connection parameter as this must not be changed on a Teradata connection.
+   Setting connection attributes like timezone would make subsequent queries behave unexpectedly."
+  [{sql :query, params :params, remark :remark} timezone connection]
+  (let [sql              (s/replace (str "-- " remark "\n" (hx/unescape-dots sql)) "OFFSET" "")
+        statement        (into [sql] params)
+        [columns & rows] (jdbc/query connection statement {:identifiers    identity, :as-arrays? true
+                                                           :read-columns   (#'metabase.driver.generic-sql.query-processor/read-columns-with-date-handling timezone)})]
+    {:rows    (or rows [])
+     :columns columns}))
+
+(defn- run-query-without-timezone [driver settings connection query]
+  (#'metabase.driver.generic-sql.query-processor/do-in-transaction connection (partial run-query query nil)))
+
+(defn- execute-query
+"Process and run a native (raw SQL) QUERY."
+  [driver {:keys [database settings], query :native, :as outer-query}]
+  (let [query (assoc query :remark (qputil/query->remark outer-query))]
+    (#'metabase.driver.generic-sql.query-processor/do-with-try-catch
+      (fn []
+        (let [db-connection (sql/db->jdbc-connection-spec database)]
+          (run-query-without-timezone driver settings db-connection query))))))
+
+(def TeradataISQLDriverMixin
```

Comment:
There's no reason to have a separate "mixin" here with a partial implementation of some of the driver methods. We only do that for the Postgres driver because the Redshift driver uses the same implementations. Since no other drivers are going to be using these Teradata implementations there's no need to make them available like this

Reply:
ok